docs/src/components/CustomAudioMeter.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+
+export interface ICustomAudioMeterProps {
+    audioMonitor: any
+}
+
+export class CustomAudioMeter extends React.PureComponent<ICustomAudioMeterProps, {}> {
+    private audioMeter: HTMLDivElement;
+
+    constructor(props: ICustomAudioMeterProps) {
+        super(props);
+
+        this.refMeter = this.refMeter.bind(this);
+    }
+
+    public componentDidMount() {
+        const { audioMonitor } = this.props;
+
+        audioMonitor.on('volume_change', (volume: number) => {
+            if (this.audioMeter) {
+                // NOTE(yunsi): volume ranges from -100 to 0.
+                // TODO(yunsi): Find a better algorithm to convert db to gain.
+                const audioGain = Math.pow(10, (volume / 20));
```

Comment:
Shall we use setState instead of operating on UIElement directly?

Reply:
https://stackoverflow.com/questions/23530716/react-how-much-can-i-manipulate-the-dom-react-has-rendered

Reply:
Added a TODO